### ISSUE TITLE
Network Interface Name is Not Accepted

### ISSUE THREAD

**Issue by BillionPenny** [MrAsterisco/homebridge-hisense-tv#5]
**Describe Your Problem:**
Upon running `python3 hisensetv.py <TV_IP_ADDRESS> --authorize --ifname <NETWORK_INTERFACE_NAME>`, with the correct network interface name, the script returns with Unknown network interface, naming the network interface.

I have looked everywhere for the network identifier for my iMac and the only possible one is "en4".

This is on the "Setting up the TV" step of the README, so no Homebridge logs nor plugin config is relevant (I think? -- happy to provide what I can if it's necessary). I'm on macOS, and used `network setup -listallhardwareports` to get the correct network interface. It's very possible that I've messed up a few other things in my setup; I've attached screenshots with the input and output.

I have no idea what I'm doing at this point (relatively new to Python) so any help would be greatly appreciated :)

**Logs:**

```
N/A (it is not inside of Homebridge)
```

**Plugin Config:**

```
N/A as the plugin has not been configured yet
```

**Screenshots:**
Here's what I'm typing in to the command line:
![image](https://user-images.githubusercontent.com/64502049/116226442-518c1500-a796-11eb-8464-50f4c6c47b3e.png)

And here's what it results in:
![image](https://user-images.githubusercontent.com/64502049/116226577-7bddd280-a796-11eb-9ef2-994bbea3e144.png)

**Environment:**

* **Plugin Version**:
* **Homebridge Version**: v1.3.4
* **Node.js Version**: v14.16.0
* **NPM Version**: v6.14.11
* **Operating System**: macOS

**Comment by chinedu40** [comment]
Do you use an ethernet or wifi connection?

Are you able to paste the output of "ip a" command on the machine running the plugin?

**Comment by BillionPenny** [comment]
I use a wifi connection.

"ip a" doesn't work as I'm running macOS and I believe that's a Linux command. I'm not sure what the Mac equivalent of ip a is.

`command not found: ip`

**Comment by chinedu40** [comment]
Hi usually on MacOS wifi is "en0". You can check this using ifconfig -L and then match your ip address with the network interfaces listed.

**Comment by BillionPenny** [comment]
So that's the thing. My wifi is "en4" (but I did try en0 as well in the command) but in all cases the setup command says that network interface is unknown. (see my second screenshot in the original post).

**Comment by chinedu40** [comment]
Unfortunately, where you put the screenshot is blank. Could you copy and paste the output of the command form terminal

**Comment by BillionPenny** [comment]
Oh sorry, my bad. It looks fine on my end so I assumed it was working.
Here's the input (not the real MAC address, but that's the format):
`python3 /usr/local/lib/node_modules/homebridge-hisense-tv-remotenow/bin/hisensetv.py 40:C0:7A:36:F0:89 --authorize --ifname en4`

and here's the output:

`Traceback (most recent call last):
  File "/usr/local/lib/node_modules/homebridge-hisense-tv-remotenow/bin/hisensetv/__init__.py", line 64, in get_mac_address
    info = fcntl.ioctl(s.fileno(), 0x8927, struct.pack('256s', bytes(ifname, 'utf-8')[:15]))
OSError: [Errno 6] Device not configured

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/node_modules/homebridge-hisense-tv-remotenow/bin/hisensetv.py", line 6, in <module>
    __main__.main()
  File "/usr/local/lib/node_modules/homebridge-hisense-tv-remotenow/bin/hisensetv/__main__.py", line 112, in main
    tv = HisenseTv(
  File "/usr/local/lib/node_modules/homebridge-hisense-tv-remotenow/bin/hisensetv/__init__.py", line 119, in __init__
    self._mac = get_mac_address(network_interface)
  File "/usr/local/lib/node_modules/homebridge-hisense-tv-remotenow/bin/hisensetv/__init__.py", line 67, in get_mac_address
    raise HisenseTvError("Unknown network interface: " + ifname)
hisensetv.HisenseTvError: Unknown network interface: en4`

**Comment by chinedu40** [comment]
So first thing that looks wrong, the command should be structured like this python3 .../hisensetv.py --authorize IP_ADDRESS_OF_TV --ifname en4. Please paste the output of that.

**Comment by BillionPenny** [comment]
Oh right. I thought the readme had said MAC address. I'll try the new command.

Here's the output:

Traceback (most recent call last):
  File "/usr/local/lib/node_modules/homebridge-hisense-tv-remotenow/bin/hisensetv/__init__.py", line 64, in get_mac_address
    info = fcntl.ioctl(s.fileno(), 0x8927, struct.pack('256s', bytes(ifname, 'utf-8')[:15]))
OSError: [Errno 6] Device not configured

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/node_modules/homebridge-hisense-tv-remotenow/bin/hisensetv.py", line 6, in <module>
    __main__.main()
  File "/usr/local/lib/node_modules/homebridge-hisense-tv-remotenow/bin/hisensetv/__main__.py", line 112, in main
    tv = HisenseTv(
  File "/usr/local/lib/node_modules/homebridge-hisense-tv-remotenow/bin/hisensetv/__init__.py", line 119, in __init__
    self._mac = get_mac_address(network_interface)
  File "/usr/local/lib/node_modules/homebridge-hisense-tv-remotenow/bin/hisensetv/__init__.py", line 67, in get_mac_address
    raise HisenseTvError("Unknown network interface: " + ifname)
hisensetv.HisenseTvError: Unknown network interface: en4

**Comment by chinedu40** [comment]
Sorry to ask really silly things. If you replace python3 with python 3.8 does that make a difference?
Could you post the output of network setup -listallhardwareports

**Comment by BillionPenny** [comment]
No it's no problem, honestly my mistakes are silly ones because I skipped steps or something.
Using Python 3.8 doesn't make a difference.
And here's the network setup:

```
Hardware Port: Ethernet
Device: en0
Ethernet Address: 10:dd:b1:a2:8c:35

Hardware Port: Wi-Fi
Device: en4
Ethernet Address: 8c:2d:aa:43:3d:87

Hardware Port: Bluetooth PAN
Device: en3
Ethernet Address: 8c:2d:aa:43:3d:88

Hardware Port: Thunderbolt 1
Device: en1
Ethernet Address: 82:0d:5a:63:04:00

Hardware Port: Thunderbolt 2
Device: en2
Ethernet Address: 82:0d:5a:63:04:01

Hardware Port: Thunderbolt Bridge
Device: bridge0
Ethernet Address: 82:0d:5a:63:04:00
```

**Comment by chinedu40** [comment]
Try authorising your mac using this https://github.com/newAM/hisensetv which is the origin of this homebridge plugin, there seems to be an issues is MacOS where the --ifname argument does not work

**Comment by MrAsterisco** [comment]
@chinedu40 thanks for jumping on this!

@BillionPenny I have tested and used the plugin on macOS a couple of times and it worked. To be honest, I have no clue why the network interface name is not accepted in your case.

The problem with the original Python script is that it always defaults to a generic MAC address, instead of using your own. This means that you can't have two devices (i.e. two instances of Homebridge) connected to the same TV, as they count as one device to the HiSense software (because they have the same MAC address).

In other words, the original Python script will work in your situation, but it won't help with the plugin, because it authenticates a different device.

I opened an issue on my fork of the Python script, as this seems to be an issue on that side: https://github.com/MrAsterisco/hisensetv/issues/1

**Comment by BillionPenny** [comment]
@MrAsterisco Thanks so much for your response! In this case should I just try the original script as mentioned by chinedu? Or is it a different issue that means that I won't be able to use the plugin until something else is fixed?

I'll mention again that I'm very ... unskilled in this field. Most of what I have on Homebridge is the result of serious bodging haha

**Comment by MrAsterisco** [comment]
@BillionPenny Ahaha, don't worry. I'll have to take a look into it, unless @chinedu40 is faster than me. You can still test with the original Python plugin, just to make sure that there are no other issues. You can also report the result of this test in the other issue, if you'd like.

**Comment by BillionPenny** [comment]
@MrAsterisco All good then. I'll give the original one a shot later when I have some time and let you know what the result is :)
